mmf/trainers/core/training_loop.py

```diff
@@ -152,3 +144,19 @@ def _extract_loss(self, report: Dict[str, Any]) -> Tensor:
         loss_dict = report.losses
         loss = sum([loss.mean() for loss in loss_dict.values()])
         return loss
+
+    def _calculate_max_updates(self):
+        max_updates = self.training_config.max_updates
+        max_epochs = self.training_config.max_epochs
+        if max_updates is None and max_epochs is None:
+            raise Exception("Neither max_updates nor max_epochs is specified.")
```

Comment:
Generally, it is always better to raise Exceptions of specific types so that someone developing on top can catch only the exceptions they care about. In this case, it will be a ValueError.